scripts/generate-artifacts.ps1

```diff
@@ -38,34 +39,37 @@ if ($artifactContainer) {
 }
 
 # generation of artifacts
-Write-Output "Generating artifacts..."
+Write-Output "Generating artifacts"
 
-if (Test-Path -Path "$PWD/$spreadsheetName" -ErrorAction SilentlyContinue) { 
-    Write-Output "Pricing calculations found - OK"
-} else { 
-    throw "Unable to find exported pricing calculations file ($spreadsheetName). Please make sure that you copied the file to the repository folder."
+Write-Output "Checking if temp folder exists"
+if (-not (Test-Path "$tempFolderPath")) { 
+    Write-Output "Temp folder does not exist, creating..."
+    New-Item -ItemType Directory -Path $tempFolderPath
 }
 
-Write-Output "Uploading spreadsheet"
+Write-Output "Exporting resources template"
+Export-AzResourceGroup -ResourceGroupName $rgName -Path "$tempFolderPath/$resourcesTemplateName" -Force
+
+Write-Output "Uploading resources template"
 $ResourcesTemplateBlob = @{
-    File             = "$PWD/$spreadsheetName"
+    File             = "$tempFolderPath/$resourcesTemplateName"
     Container        = $containerName
-    Blob             = "$taskName/$spreadsheetName"
+    Blob             = "$taskName/$resourcesTemplateName"
     Context          = $storageAccount.Context
     StandardBlobTier = 'Hot'
 }
 $blob = Set-AzStorageBlobContent @ResourcesTemplateBlob -Force
 
-Write-Output "Generating a SAS token for the artifact"
+Write-Output "Generating a SAS token for the template artifact"
 $date = Get-Date
 $date = $date.AddDays(30) 
-$resourcesTemplateSaSToken = New-AzStorageBlobSASToken -Container $containerName -Blob "$taskName/$spreadsheetName" -Permission r -ExpiryTime $date -Context $storageAccount.Context
-$pricingCalculationsURL = "$($blob.ICloudBlob.uri.AbsoluteUri)?$resourcesTemplateSaSToken"
+$resourcesTemplateSaSToken = New-AzStorageBlobSASToken -Container $containerName -Blob "$taskName/$resourcesTemplateName" -Permission r -ExpiryTime $date -Context $storageAccount.Context
```

Comment:
The SAS token is generated with a 30-day expiry. Ensure this duration aligns with your security and access policies.